cips/cip-tx-limits.md

```diff
@@ -0,0 +1,44 @@
+| cip | TBD(27) |
+| - | - |
+| title | Block limits for number of PFBs/non-PFBs and transaction sizes |
+| description | Set limits for number of PFBs and non-PFBs per block and transaction size |
+| author | Josh Stein ([@jcstein](https://github.com/jcstein)) |
+| discussions-to |  <https://forum.celestia.org/t/cip-limit-number-of-pfbs-and-non-pfbs-per-block-increase-transaction-size-limit/1843> |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-10-16 |
+
+## Abstract
+
+This CIP proposes to set limits for the number of PFBs and non-PFBs per block and set the limit for transaction size. Concretely, the proposal is to set the limits to 600 PFBs and 200 non-PFB transactions per block and set the transaction size limit to 2MiB.
+
+## Specification
+
+1. The number of PFBs per block is limited to 600 by setting `PFBTransactionCap` to 600. `PFBTransactionCap` is the maximum number of PFB messages a block can contain.
+
+1. The number of non-PFB transactions per block is limited to 200 by setting `NonPFBTransactionCap` to 200. `NonPFBTransactionCap` is the maximum number of SDK messages, aside from PFBs, that a block can contain.
```

Comment:
[optional] I think it's worth calling out that these limits aren't actually enforced. They're limits used by the celestia-app implementation but a validator could hypothetically modify prepare proposal, run that binary and create blocks with more PFBs or non-PFBs.

[nit] the constants are poorly named b/c the thing being limited is the number of messages in a block not the number of transactions. Should we change the implementation and the specs here? cc: @rach-id

Reply:
https://github.com/celestiaorg/CIPs/pull/220/commits/2d60926e4786ebed5c909c8176332bef340c4609

Reply:
@rootulp yes sure. want me to open a PR or you do it?

Reply:
https://github.com/celestiaorg/CIPs/pull/220/commits/8fbee0ed2f37bc2c8c97be877dc007c92b85bf55 updated to new variables, if these don't get included with v3.0.0-rc, I will revert to names `NonPFBTransactionCap` and `PFBTransactionCap`